modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/AttachmentResourceTest.java

```diff
@@ -6,14 +6,367 @@
+	}
+
+	@Override
+	protected Attachment
+			testDeleteAttachmentByExternalReferenceCode_addAttachment()
+		throws Exception {
+
+		return attachmentResource.postProductByExternalReferenceCodeAttachment(
+			_cProduct.getExternalReferenceCode(), randomAttachment());
+	}
+
+	@Override
+	protected Attachment
+			testGetAttachmentByExternalReferenceCode_addAttachment()
+		throws Exception {
+
+		return attachmentResource.postProductByExternalReferenceCodeAttachment(
+			_cProduct.getExternalReferenceCode(),
+			_randomAttachmentWithFileEntryExternalReferenceCode());
+	}
+
+	@Override
+	protected Attachment
+			testPatchAttachmentByExternalReferenceCode_addAttachment()
+		throws Exception {
+
+		return attachmentResource.postProductByExternalReferenceCodeAttachment(
+			_cProduct.getExternalReferenceCode(), randomAttachment());
+	}
+
+	@Override
+	protected Attachment
+			testPostProductByExternalReferenceCodeAttachment_addAttachment(
+				Attachment attachment)
+		throws Exception {
+
+		return attachmentResource.postProductByExternalReferenceCodeAttachment(
+			_cProduct.getExternalReferenceCode(),
+			_randomAttachmentWithFileEntryExternalReferenceCode());
+	}
+
+	@Override
+	protected Attachment testPostProductIdAttachment_addAttachment(
+			Attachment attachment)
+		throws Exception {
+
+		return attachmentResource.postProductIdAttachment(
+			_cProduct.getCProductId(), randomAttachment());
+	}
+
+	private Attachment _randomAttachmentWithFileEntryExternalReferenceCode()
+		throws Exception {
+
+		FileEntry fileEntry = _dlAppLocalService.addFileEntry(
+			null, TestPropsValues.getUserId(), testGroup.getGroupId(),
+			DLFolderConstants.DEFAULT_PARENT_FOLDER_ID,
+			TempFileEntryUtil.getTempFileName(
+				RandomTestUtil.randomString() + ".txt"),
+			ContentTypes.TEXT_PLAIN, RandomTestUtil.randomString(),
+			StringPool.BLANK, StringPool.BLANK, StringPool.BLANK,
+			new ByteArrayInputStream(RandomTestUtil.randomBytes()), 0, null,
+			null, null, _serviceContext);
+
+		Group group = GroupLocalServiceUtil.getGroup(fileEntry.getGroupId());
+
+		return new Attachment() {
+			{
+				cdnEnabled = false;
+				cdnURL = RandomTestUtil.randomString();
+				externalReferenceCode = RandomTestUtil.randomString();
+				fileEntryExternalReferenceCode =
+					fileEntry.getExternalReferenceCode();
+				fileEntryGroupExternalReferenceCode =
+					group.getExternalReferenceCode();
+				galleryEnabled = true;
+				id = RandomTestUtil.randomLong();
+				neverExpire = true;
+				priority = RandomTestUtil.randomDouble();
+				title = HashMapBuilder.put(
+					"en_US", RandomTestUtil.randomString(5)
+				).build();
+				type = RandomTestUtil.randomInt();
+			}
+		};
+	}
+
+	@DeleteAfterTestRun
+	private CPDefinition _cpDefinition;
+
+	@DeleteAfterTestRun
+	private CProduct _cProduct;
+
+	@Inject
+	private DLAppLocalService _dlAppLocalService;
+
+	private ServiceContext _serviceContext;
+
+	@DeleteAfterTestRun
+	private User _user;
+
```

Comment:
I think we need to add the usual _testPost****WithFileEntryExternalReferenceCode because otherwise we don't have any assert that checks if the id returned is the same of the one related to the ERC